### ISSUE TITLE
XML Comments

### ISSUE THREAD

**Issue by premek** [jOOQ/jOOX#144]
Allowing to work with XML comments would be a nice enhancement, however I'm not sure how would it fit into the current API.

Comment nodes are parsed by the XML parser but are discarded here
https://github.com/jOOQ/jOOX/blob/aee98f273d8f63b7a85f88b3df6d2f6ea0e3b23a/jOOX/src/main/java/org/joox/Elements.java#L70
So methods like children() gets only the elements, not the comments. (Which is correct by default but would be nice to be able to get commments too maybe)

Possible use case example:
- delete selected nodes from xml file including the comment right before them

**Comment by lukaseder** [comment]
Yes, that would be interesting, but jOOQ's `Match` type explicitly models an `Element`, keeping things simple, but reducing the set of possible functionalities. For instance, text of cdata nodes are also not possible right now, at least not if they're interleaved with other elements in a document style XML.

Perhaps there's a solution by creating a new super type of `Match` and then retrofit the existing `Match` to be an instance of that super type specific to elements...

But right now, this won't be a priority for jOOX...
